### PR TITLE
refactor(cli/inspector): make server optional

### PR DESCRIPTION
**Pull request by caspervonb**
This makes inspector registration with the server with optional and explicit to allow for inspectors to exist without spinning up the server.

As a side effect of the server being explicitly passed around it also makes it possible to bind multiple servers (Not likely we'll ever need that, but it is possible).

### CODE REVIEW

**Comment by caspervonb** [comment]
Ties into #7473

**Comment by bartlomieju** [comment]
@caspervonb so `--coverage` will not print `Debugger listening on...` message?

**Comment by caspervonb** [comment]
> @caspervonb so --coverage will not print Debugger listening on... message?

In a follow up will, this only makes the two separable but doesn't change the behaviour of current flags configuration/parsing.

File: cli/worker.rs [hunk 1]
```diff
@@ -161,7 +162,12 @@ impl Worker {
         .inspect
         .or(global_state.flags.inspect_brk)
         .filter(|_| !is_internal)
-        .map(|inspector_host| DenoInspector::new(&mut isolate, inspector_host))
+        .map(|inspector_host| {
+          DenoInspector::new(
+            &mut isolate,
+            Some(Arc::new(InspectorServer::new(inspector_host))),
+          )
+        })
```
Comment (bartlomieju) [inline cli/worker.rs:170]: This call will result in multiple servers being created

**Review by bartlomieju — CHANGES_REQUESTED** [review]

**Review by bartlomieju — APPROVED** [review]
LGTM, nice cleanup @caspervonb

**Pull request merged**